sdk/go/go.mod

```diff
@@ -30,6 +30,8 @@ require (
 	google.golang.org/grpc v1.64.0
 )
 
+replace go.opentelemetry.io/otel/sdk/log => go.opentelemetry.io/otel/sdk/log v0.3.0
```

Comment:
:cry:

While we're doing this, should we also pin any *other* OTEL deps? Or do you think it's fine to keep this as is?

Reply:
This one is needed because it's pre-1.0 and they're still working out the API.

Theoretically this is a problem for any Go SDK + modules dependency that makes a breaking change without a major version bump. But if we expand the scope to include all pre-1.0 modules, suddenly that includes quite a few things (`Khan/genqlient`, `99designs/gqlgen`, `golang.org/x/sync`, `golang.org/x/exp`).

Arguably this is another crufty symptom of carrying around the source code for `querybuilder` and `telemetry` - pretty sure if they were external packages they could just have normal dependencies that would always be respected.

Or maybe we could just give each package its own `go.mod`? 🤔 

In the short term we should at least pin the rest of the OTel SDK logging related packages since they're most likely to break again, so I'll do that.

Reply:
That's true, but I think we don't need to worry so much about that, because `dagger.io/dagger` will have its own set of dependencies in its own `go.mod`, and those should still be respected - we don't have to worry about the `dagger.io/dagger` dependencies getting bumped out-of-band.

At least, that's my understanding of how Go module dependencies work. The Dagger module case is unique because it vendors the relevant telemetry code directly, so it's subject to the module's own `go.mod`. The `replace` rules are just in `sdk/go/go.mod` so we can have it all in one place and carry them over to each module's `go.mod` and overrule any accidental bumps.